scripts/daq_utils.py

```diff
@@ -0,0 +1,226 @@
+
+
+class DaqManager:
+    def __init__(self, opts):
+        self.opts = opts
+        self.verbose = False
+        for o, a in self.opts:
+            if o in ('-v', '--verbose'):
+                self.verbose = True
+        self.hutch = call_subprocess("get_info", "--gethutch")
+        self.user = self.hutch+'opr'
+        self.sbman = SbatchManager(self.user)
+        self.scripts_dir = f'/reg/g/pcds/dist/pds/{self.hutch}/scripts'
+        self.set_cnf_file_and_platform()
+
+    def isdaqbatch(self, quiet=False):
+        if self.hutch in DAQBATCH_HUTCHES:
+            if not quiet: print('true')
+            return True
+        else:
+            if not quiet: print('false')
+            return False
+
+    def isvaliduser(self):
+        if getpass.getuser().find('opr') > -1:
+            return True
+        return False
+
+    def waitfor(self, subcmd):
+        if subcmd == 'stop':
+            daq_host = self.wheredaq(quiet=True)
+            if daq_host is not None:
+                if self.isdaqbatch(quiet=True):
+                    for i_retry in range(MAX_RETRIES):
+                        daq_host = self.wheredaq(quiet=True)
+                        if daq_host is None:
+                            break
+                        print(f'wait for daqbatch to stop #retry: {i_retry}')
+                        time.sleep(1)
+                else:
+                    print(f'the DAQ did not stop properly (DAQ HOST: {daq_host}), exit now and try again or call your POC or the DAQ phone')
+
+    def set_cnf_file_and_platform(self):
+        self.cnf_file = f'{self.hutch}.py'
+        
+        cc = {'platform': None}
+        configfilename = os.path.join(self.scripts_dir, self.cnf_file)
+        try:
+            exec(compile(open(configfilename).read(), configfilename, 'exec'), {}, cc)
+            if type(cc['platform']) == type('') and cc['platform'].isdigit():
+                rv = int(cc['platform'])
+        except:
+            print('deduce_platform Error:', sys.exc_info()[1])
+            raise
+        self.platform = rv
+
+    def wheredaq(self, quiet=False):
+        """ Locate where the daq is running.
+        
+        For daqbatch, we use slurm to check if the hutch user is running control_gui.
+        """
+        daq_host = None
+        # Use control_gui job name to locate the running host for the daq
+        job_details = self.sbman.get_job_info()
+        if 'control_gui' in job_details:
+            daq_host = job_details['control_gui']['nodelist']
+        
+        if not quiet:
+            if daq_host is None: 
+                print(f'DAQ is not running in {self.hutch}')
+            else:
+                print(f'DAQ is running on {daq_host}')
+        return daq_host
+    
+    def calldaq(self, subcmd, daq_host=None):
+        prog = 'daqbatch'
+        cmd = f'pushd {self.scripts_dir}'+' > /dev/null;'
+        cmd += f'source {os.path.join(self.scripts_dir, "setup_env.sh")}'+';'
```

Comment:
To confirm my understanding: The way we invoke these `daqbatch` commands is by first
* sourcing the `setup_env.sh` script, which activates a `ps-x.x.x` conda environment
* Calling the `daqbatch` command line entrypoint and its subcommands

This environment has access to `psdaq`, so is there a reason we don't just activate this environment once at the beginning of the the script and directly call on `psdaq`'s utilities?  There's a lot of brittle-looking runtime script generation and execution, which might be avoidable

Reply:
To my understand psdaq talks to a DAQ that is already running. We are trying to start the DAQ here.

Reply:
This method is used for all the daq commands, as written in this code (start, stop, restart, etc)